enhancements/aws-ipv6-dev-env.md

```diff
@@ -0,0 +1,118 @@
+---
+title: AWS-IPv6-Dev-Env
+authors:
+  - "@russellb"
+reviewers:
+  - TBD
+approvers:
+  - TBD
+creation-date: 2019-10-24
+last-updated: 2019-10-24
+status: implementable
+---
+
+# aws-ipv6-dev-env
+
+## Release Signoff Checklist
+
+- [x] Enhancement is `implementable`
+- [ ] Design details are appropriately documented from clear requirements
+- [ ] Test plan is defined
+- [ ] Graduation criteria for dev preview, tech preview, GA
+- [ ] User-facing documentation is created in [openshift/docs]
+
+## Summary
+
+While Kubernetes itself includes support for IPv6, the OKD distribution does
+not fully support it yet.  To enable developers across the many components that
+make up OKD, this proposal is to add support for enabling IPv6 for an AWS
+cluster.  This will also be used to create a CI job that can exercise IPv6
+support across OKD.
+
+## Motivation
+
+Supporting IPv6 across OKD will require collaboration and fixes across many
+components.  Many developers are already using to doing development against AWS
+clusters, so supporting AWS as an IPv6 development environment should make it
+easier for people to jump in to help debug and fix IPv6 issues.
+
+### Goals
+
+* Make creating an IPv6 development environment as easy as creating an AWS
+  cluster, by just setting one additional option to turn it on.
+* Support OKD in reaching a milestone of supporting a single stack IPv6 control
+  plane (IPv6 only).
+
+### Non-Goals
+
+* Reaching full support for IPv6 on AWS.  While the work will get us closer,
+  that is not the primary goal here.  The primary goal is to help make it
+  easier for all teams to exercise IPv6.
+
+## Proposal
+
+A PR with an implementation and documentation can be found here:
+
+* <https://github.com/openshift/installer/pull/2555>
```

Comment:
IMO this doesn't seem right, personally the enhacement should drive the implementation not the other way around.
I have no way of commenting if the process is flawed because I don't have the process documented here.

Reply:
If you mean it would be a good idea to have more details in this doc so it can be discussed separately from the implementation ... then that makes sense

It would probably work well to take the content from the "AWS Network Environment" section of https://github.com/openshift/installer/blob/bd498521d3cd9936adaa02f700e55c57896e9067/docs/dev/aws_ipv6.md and elaborate on why that design makes sense ... and a lot of the rationale can be pulled from Russell's nice commit messages